doc/source/design/design-timestep.rst

```diff
@@ -0,0 +1,202 @@
+.. include:: ../roles.incl
+
+.. |tijk| replace:: t\ :sub:`i` :sup:`j,k`
+
+.. toctree::
+
+.. _Adaptive Time-Step Design:
+
+
+*************************
+Adaptive Time-Step Design
+*************************
+
+At the core of an Enzo-E / Cello simulation is a sequence of physics
+methods that is repeatedly applied to mesh blocks inside a loop, with
+each cycle through the methods and blocks advancing the
+simulation one global time-step. This document describes optimizations
+related to how time-stepping is performed.
+
+In a given cycle, each method / block pair has an associated maximum
+local time-step, which is computed using the
+``Method<Foo>::timestep(block)`` function. This applies the largest
+time-step that the method can safely take on that particular block.
+These local time-steps can be different for different methods, on
+different blocks, and at different points in time.
+
+The simplest time-stepping strategy is to use a constant global
+time-step for each cycle, computed as the minimum local time-step over
+all methods and blocks in the cycle. This is the approach that Enzo-E
+/ Cello used initially. While easy to implement, depending on the
+simulation characteristics it can be very inefficient.
+
+Two ways to improve time-stepping performance are to 1) allow adaptive
+time-stepping in different regions of the mesh, and 2) allow different
+time-steps for different methods.  An example of 1) is the original
+ENZO code: ENZO uses different time-steps on different mesh refinement
+levels, which can significantly improve performance on mesh
+hierarchies with many refinement levels. An example of 2) would be
+advancing the gravitational potential, which can be costly due to its
+global communication requirements, at a larger time-step than other
+methods when possible.
+
+Our plan is to optimize time-stepping in terms of both methods (via
+"super-cycling") and individual blocks (via "block-adaptive
+time-stepping").  We implement super-cycling first since it requires
```

Comment:
Does block-adaptive time-stepping mean that blocks on the same refinement level can have different timesteps?